ledger/acctupdates.go

```diff
@@ -416,6 +417,126 @@ func (au *accountUpdates) lookupKv(rnd basics.Round, key string, synchronized bo
 	}
 }
 
+func (au *accountUpdates) LookupKeysByPrefix(round basics.Round, keyPrefix string, maxKeyNum uint64) ([]string, error) {
+	return au.lookupKeysByPrefix(round, keyPrefix, maxKeyNum, true /* take lock */)
+}
+
+func (au *accountUpdates) lookupKeysByPrefix(round basics.Round, keyPrefix string, maxKeyNum uint64, synchronized bool) (resultKeys []string, err error) {
+	var results map[string]bool
+	// keep track of the number of result key with value
+	var resultCount uint64
+
+	needUnlock := false
+	if synchronized {
+		au.accountsMu.RLock()
+		needUnlock = true
+	}
+	defer func() {
+		if needUnlock {
+			au.accountsMu.RUnlock()
+		}
+		// preparation of result happens in deferring function
+		// prepare result only when err != nil
+		if err == nil {
+			resultKeys = make([]string, 0, resultCount)
+			for resKey, isValid := range results {
+				if isValid {
+					resultKeys = append(resultKeys, resKey)
+				}
+			}
+		}
+	}()
+
+	// TODO: This loop and round handling is copied from other routines like
+	// lookupResource. I believe that it is overly cautious, as it always reruns
+	// the lookup if the DB round does not match the expected round. However, as
+	// long as the db round has not advanced too far (greater than `rnd`), I
+	// believe it would be valid to use. In the interest of minimizing changes,
+	// I'm not doing that now.
+
+	for {
+		currentDBRound := au.cachedDBRound
+		currentDeltaLen := len(au.deltas)
+		offset, _err := au.roundOffset(round)
+		if _err != nil {
+			err = _err
+			return
+		}
+
+		// reset `results` to be empty each iteration
+		// if db round does not match the round number returned from DB query, start over again
+		results = map[string]bool{}
```

Comment:
Please add a comment explaining the convention that a false means a deletion of the key has been seen, which a true means the key has been seen set to an actual value. (which will also let you explain why we need `resultCount` instead of `len(results)`)

I think when most people see this, they will assume it's a simple set of results.